algebraic-types/README.md

```diff
@@ -0,0 +1,7 @@
+
+Extra tools and samples for the Bril algebraic types extension.
+
+ - `briltc`: Bril type-checker with support for algebraic types
```

Comment:
Seems like this can just be at the top level? We do have `type-infer` there, and this is similar to that but different. The fact that it supports the ADT extension is not terribly important; someday, people might extend other stuff with similar support (or extend this with support for other extensions).